src/Svg.js

```diff
@@ -0,0 +1,71 @@
+import React, { Component, forwardRef } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import './Svg.css';
+
+export class Svg extends Component {
+    static getDerivedStateFromProps(props, state) {
+        return {
+            contents: typeof props.svg === 'string' ? props.svg : state.contents,
+        };
+    }
+
+    promise = undefined;
+    state = { contents: '' };
+
+    componentDidMount() {
+        this.maybeWaitForSvg();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (this.props.svg !== prevProps.svg) {
+            this.maybeWaitForSvg();
+        }
+    }
+
+    componentWillUnmount() {
+        this.resetWaitForSvg();
+    }
+
+    render() {
+        const { svg, className, forwardedRef, ...rest } = this.props;
+        const { contents } = this.state;
+        const finalProps = {
+            ...rest,
+            className: classNames('svgWrapper', className),
+        };
+
+        return <i { ...finalProps } ref={ forwardedRef } dangerouslySetInnerHTML={ { __html: contents } } />;
```

Comment:
Since this is a component that should be fitted for any project, I would advise leaving space for passing props to the SVG element. 

The simplest example I can think of is specifying a certain svg as `role="presentation`. There's no way on the current approach for that.

Reply:
Also, why is the `<i>` wrapper necessary? It just leads to people having to do `width: 100%; height: 100%` in the svg for it to inherit the `<i>` size.

Reply:
`contents` is a string, and it starts with
```html
<svg>....
```
To use `svg` directly we would have to parse the string, remove the svg and then set the result with dangerouslySetInnerHTML in a `svg`.

Reply:
We decided to render the `svg` itself  :+1: